.github/workflows/cd.yaml

```diff
@@ -0,0 +1,33 @@
+name: CD Workflow
+
+on: 
+  push:
+    tags:
+      - "services/[a-zA-Z]+/v[0-9]+.[0-9]+.[0-9]+"
+      - "services/[a-zA-Z]+/v[0-9]+.[0-9]+.[0-9]+-*"
+  workflow_dispatch:
+
+jobs:
+  main:
+    name: Build & Publish module
+    runs-on: "ubuntu-latest"
+    steps:      
+      - name: Checkout
+        uses: actions/checkout@v4     
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - name: Extract updated module path
+        env:
+          TAG: ${{ github.event.ref }}
+        run: |
+          TAG_NO_PREFIX=$(echo "$TAG" | sed 's/^refs\/tags\///')
+          SERVICE_PATH=$(echo "$TAG_NO_PREFIX" | rev | cut -d'/' -f2- | rev)          
+          echo "SERVICE_PATH=$SERVICE_PATH" >> $GITHUB_ENV
```

Comment:
We could add a comment here explaining what is the goal with this

Reply:
Added comments